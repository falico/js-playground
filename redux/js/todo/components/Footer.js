import React from 'react'
import { VisibilityFilters } from '../actions/constants'
import FilterLink from '../containers/FilterLink'
import RouteFilterLink from '../containers/RouteFilterLink'

const Footer = () => (
  <div>
    <p>Click the filters to see the <b>top</b> list change</p>
    <div>
      Show:
      {" "}
      <FilterLink filter={VisibilityFilters.SHOW_ALL}>
        All
      </FilterLink>
      {", "}
      <FilterLink filter={VisibilityFilters.SHOW_ACTIVE}>
        Active
      </FilterLink>
      {", "}
      <FilterLink filter={VisibilityFilters.SHOW_COMPLETED}>
        Completed
      </FilterLink>
    </div>
    <div>
      <p>Click the filters to see the <b>URL & the bottom</b> list change</p>
      Show:
      {" "}
      <RouteFilterLink filter='all'>
        All
      </RouteFilterLink>
      {", "}
      <RouteFilterLink filter='active'>
        Active
      </RouteFilterLink>
      {", "}
      <RouteFilterLink filter='completed'>
        Completed
      </RouteFilterLink>
    </div>
  </div>
)

export default Footer
