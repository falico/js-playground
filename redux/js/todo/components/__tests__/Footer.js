import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import Footer from '../Footer'
import Store from '../../__mocks__/store'

describe('Footer', () => {

  it('renders', () => {
    const component = renderer.create(
      <Provider store={Store({})}>
        <Footer />
      </Provider>,
    );

    // console.log(component.debug());

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
