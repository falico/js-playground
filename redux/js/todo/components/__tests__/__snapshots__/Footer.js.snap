exports[`Footer renders 1`] = `
<p>
  Show:
   
  <button
    href="#"
    onClick={[Function]}>
    All
  </button>
  , 
  <button
    href="#"
    onClick={[Function]}>
    Active
  </button>
  , 
  <button
    href="#"
    onClick={[Function]}>
    Completed
  </button>
</p>
`;
